index.js

```diff
@@ -2,147 +2,87 @@
 'use strict';
 
 var Funnel = require('broccoli-funnel');
-var versionChecker = require('ember-cli-version-checker');
-var path = require('path');
-var fs = require('fs');
+var Concat = require('broccoli-concat');
+var Merge = require('broccoli-merge-trees');
+var WrappedStyles = require('./lib/pod-style.js');
+var ExtractNames = require('./lib/pod-names.js');
+var RemoveEmpty = require('./lib/remove-empty.js');
 
-var ComponentCssPreprocessor = require('./lib/component-css-preprocessor');
-var ComponentCssPostprocessor = require('./lib/component-css-postprocessor');
-
-function monkeyPatch(EmberApp) {
-  var upstreamMergeTrees = require('broccoli-merge-trees');
-  var p = require('ember-cli-preprocess-registry/preprocessors');
-  var preprocessCss = p.preprocessCss;
-  var preprocessMinifyCss = p.preprocessMinifyCss;
-
-  function mergeTrees(inputTree, options) {
-    var tree = upstreamMergeTrees(inputTree, options);
-
-    tree.description = options && options.description;
-
-    return tree;
-  }
-
-  EmberApp.prototype._filterAppTree = function() {
-    if (this._cachedFilterAppTree) {
-      return this._cachedFilterAppTree;
-    }
-
-    var excludePatterns = [].concat(
-      this._podTemplatePatterns(),
-      this._podStylePatterns(),
-      [
-        // note: do not use path.sep here Funnel uses
-        // walk-sync which always joins with `/` (not path.sep)
-        new RegExp('^styles/'),
-        new RegExp('^templates/'),
-      ]
-    );
-
-    return this._cachedFilterAppTree = new Funnel(this.trees.app, {
-      exclude: excludePatterns,
-      description: 'Funnel: Filtered App'
-    });
-  };
+module.exports = {
+  name: 'ember-component-css',
 
-  EmberApp.prototype._podStylePatterns = function() {
-    return this.registry.extensionsForType('css').map(function(extension) {
-      return new RegExp(extension + '$');
+  _getPodStyleFunnel() {
+    return new Funnel('app', {
+      srcDir: this._podDirectory(),
+      exclude: ['styles/**/*'],
+      include: this.allowedStyleExtentions.map(function(ext) { return new RegExp(ext + '$'); }),
+      allowEmpty: true,
+      annotation: 'Funnel (ember-component-css grab files)'
     });
-  };
+  },
 
-  EmberApp.prototype.styles = function() {
-    if (fs.existsSync('app/styles/' + this.name + '.css')) {
-      throw new SilentError('Style file cannot have the name of the application - ' + this.name);
+  _concatenatedPodStyles(wrappedStyles) {
+    var concatenatedStyles = [];
+
+    for (var i = 0; i < this.allowedStyleExtentions.length; i++) {
```

Comment:
three are about 5 css preprocessing extensions? .scss, .sass, .less, . styl, and maybe one more?